### PR TITLE
Metric adjustments

### PR DESCRIPTION
**Pull request by cammace**

### CODE REVIEW

File: navigation/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigation.java [hunk 1]
```diff
@@ -597,6 +598,7 @@ public OffRoute getOffRouteEngine() {
 
   @Override
   public void onProgressChange(Location location, RouteProgress routeProgress) {
+    Timber.v("Arrived event occurred");
```
Comment (ericrwolfe) [inline navigation/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigation.java]: Won't all of the logic in this `onProgressChange` callback get called multiple times per navigation session?

File: navigation/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigationOptions.java [hunk 2]
```diff
@@ -76,6 +80,7 @@ public static Builder builder() {
       .snapToRoute(true)
       .manuallyEndNavigationUponCompletion(false)
       .defaultMilestonesEnabled(true)
+      .metersRemainingTillArrival(NavigationConstants.METERS_REMAINING_TILL_ARRIVAL)
```
Comment (ericrwolfe) [inline navigation/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigationOptions.java]: Do we want to open this up as a new navigation option as opposed to using the maneuver zone radius for now?

File: navigation/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigation.java (hunk 1 above)
Comment (cammace) [reply navigation/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigation.java]: Nope, it only gets invoked when close to the destination: https://github.com/mapbox/mapbox-navigation-android/blob/master/navigation/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationEventDispatcher.java#L116-L118

File: navigation/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigationOptions.java (hunk 2 above)
Comment (cammace) [reply navigation/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigationOptions.java]: I think it makes sense to have this as a separate option and has little to no cons to leaving it like this.

File: navigation/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigation.java (hunk 1 above)
Comment (ericrwolfe) [reply navigation/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigation.java]: Ahhh, I was confused by the naming. Maybe we should change the internal `onProgressChange` method to `onArrival`.

**Pull request merged**